src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Features;
+using System;
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    /// <summary>
+    /// Defines extensions for part relationships
+    /// </summary>
+    public static class OpenXmlPartExtensions
+    {
+        internal static OpenXmlPart InitPart(this OpenXmlPartContainer parent, OpenXmlPart childPart, PartTypeInfo partType, string? relId = null)
```

Comment:
This should be generic for the childpart so it you don't have to cast. Also should be private